sympy/functions/elementary/piecewise.py

```diff
@@ -276,6 +282,25 @@ def eval(cls, *_args):
 
             current_cond.add(cond)
 
+            #   Does not add intervals which are already covered entirely by
```

Comment:
Much of this can already be handled with `.as_expr_set_pairs(x)`.
You would only keep expr/cond pairs that have a non-EmptySet interval:

```python
>>> Piecewise(
...         (1, x >= 1),
...         (2, x >= 2),
...         (3, x >= 3),
...         (4, x >= 4),
...         (5, x >= 5),
...         (6, True)).as_expr_set_pairs()
[(1, Interval(1, oo)), (2, EmptySet()), (3, EmptySet()), (4, EmptySet()), (5, EmptySet()), (6, Interval.open(-oo, 1))]
>>> newargs = [a for a, (_, s) in zip(eq.args, _) if s is not S.EmptySet]
```

It would be good to add this diff to check whether the conditions are univariate, however:
```diff
diff --git a/sympy/functions/elementary/piecewise.py b/sympy/functions/elementar
y/piecewise.py
index edd4dab..2c79a87 100644
--- a/sympy/functions/elementary/piecewise.py
+++ b/sympy/functions/elementary/piecewise.py
@@ -961,7 +961,12 @@ def as_expr_set_pairs(self, domain=S.Reals):
         exp_sets = []
         U = domain
         complex = not domain.is_subset(S.Reals)
+        cond_free = set()
         for expr, cond in self.args:
+            cond_free |= cond.free_symbols
+            if len(cond_free) > 1:
+                raise NotImplementedError(filldedent('''
+                    multivariate conditions are not handled.'''))
             if complex:
                 for i in cond.atoms(Relational):
                     if not isinstance(i, (Equality, Unequality)):
```
This tests as
```python
>>> Piecewise((x,x<0),(0,y>1)).as_expr_set_pairs()
Traceback (most recent call last):
...
NotImplementedError:
multivariate conditions are not handled.
```

Reply:
@smichr I completely changed the code to make this an optional functionality only.